pkg/distributor/distributor.go

```diff
@@ -177,26 +179,76 @@ func (d *Distributor) stopping(_ error) error {
 	return services.StopManagerAndAwaitStopped(context.Background(), d.subservices)
 }
 
-func (d *Distributor) Push(ctx context.Context, req *connect.Request[pushv1.PushRequest]) (*connect.Response[pushv1.PushResponse], error) {
+func (d *Distributor) Push(ctx context.Context, grpcReq *connect.Request[pushv1.PushRequest]) (*connect.Response[pushv1.PushResponse], error) {
```

Comment:
Why did you need to change this ?

Reply:
I want to be able to Push a parsed pprof and I dont see how I could do it with grpc.

I want to be able Push `[]byte` with grpc and `*pprof.Profile` with a direct function call
So there's new `ProfileSample` which has both and sometimes `RawProfile []byte` is nil (for a case when pprof was not parsed from bytes, i.e. jfr)

I also want to have a single function processing both cases avoiding code duplication.

Do you think there may be other ways to implement this? I'd like to improve it

Reply:
Not sure but I'll provide feedback, I think it makes senses.

Reply:
One thing that could help either having 2 commits, or 2 prs too for both change, the jfr improvement and the parsed pprof change.

Reply:
Yes, let's do two PR's!

Reply:
But I need an idea of how to implement the second one before I start

Reply:
reverted changes in distributor